processor/src/decoder/mod.rs

```diff
@@ -268,6 +275,33 @@ impl Decoder {
         self.trace.program_hash()
     }
 
+    /// Returns an operation to be executed at the specified clock cycle. Only applicable in debug mode.
+    pub fn get_operation_at(&self, clk: usize) -> Operation {
+        self.operations[clk]
+    }
+
+    /// Returns the asmop decorator at the specified clock cycle.
+    /// - Returns asmop decorator if one exists at the specified clock cycle.
+    /// - Returns None otherwise
+    pub fn get_asmop_at(&mut self, clk: usize) -> Option<AsmOpInfo> {
+        if self.asmop_idx < self.asmop_list.len() && self.asmop_list[self.asmop_idx].0 == clk {
+            self.asmop_idx += 1;
+            Some(self.asmop_list[self.asmop_idx - 1].1.clone())
+        } else {
+            None
+        }
+    }
```

Comment:
Since this is a getter, we shouldn't be using a mutable reference for `self`.

Reply:
Makes sense. Removed.